src/entity/light.cpp

```diff
@@ -0,0 +1,26 @@
+#include <light.h>
+
+Light::Light() : location(Point()), intensity(Intensity()) {}
+
+Light::Light(Point location, Intensity intensity) : location(location), intensity(intensity) {}
+
+int Light::Intensity::clamp(int value) {
+    if (value < 0)
+        return 0;
+    else if (value > 255)
+        return 255;
+    else
+        return value;
```

Comment:
legal a ideia, mas acho melhor a gente gerar um erro mesmo, pq no meio do projeto a gente pode identificar mais fácil.
```suggestion
    assert(value >= 0 && value <= 255);
    return value;
```